### PR TITLE
Add standalone operational mode

### PR DESCRIPTION
**Pull request by HotelCalifornia**
This needs testing. I had a dubious rig set up, but then I lost the toolchain. Oh 🐳.

### CODE REVIEW

**Comment by HotelCalifornia** [comment]
Testing checklist:

- [ ] code runs in auto with tether
- [ ] code runs without waiting for vexnet (without tether)

File: src/supervisor.c [hunk 1]
```diff
@@ -194,7 +194,7 @@ void svSynchronize() {
 	// HELLO message!
 	kwait(50);
 	print("\r\nPowered by PROS " FW_VERSION "\r\n" FW_DISCLAIMER
-		"\r\nPROS (C)2011-2014 Purdue ACM SIGBOTS\r\n");
+		"\r\nPROS (C)2011-2017 Purdue ACM SIGBOTS\r\n");
```
Comment (HotelCalifornia) [inline src/supervisor.c]: I know it's not strictly related, but while we're here and editing `supervisor.c`...

File: src/supervisor.c [hunk 2]
```diff
@@ -137,7 +137,9 @@ void _svNextByte() {
 
 // standaloneModeEnable - enable standalone operation
 void standaloneModeEnable() {
-	svInitFlags = 0x1;
```
Comment (edjubuh) [inline src/supervisor.c]: This didn't work because svInit runs before initIO, so the actual TX SPI buffer never gets updated.

**Comment by baylessj** [comment]
The final version of this feature implements standalone mode by setting a flag in the outgoing SPI communication buffer between the user and master processors to specify to the master processor that the controller should run in standalone mode. Because this process occurs during the `initializeIO()` function, there is a very brief moment of time between the SPI communication's initialize at boot-up and when the `initializeIO()` function runs where the controller has not yet entered standalone mode (evidenced by typical "searching for a VEXNet connection" lights), but as soon as the `initializeIO()` function runs then the VEXNet light drops out and the controller enters standalone mode.

Succesful test cases include:
 
- `standaloneModeEnable()` called in `initializeIO()`, running tethered.
- `standaloneModeEnable()` called in `initializeIO()`, running standalone (no VEXNet/USB connection)
- no call to `standaloneModeEnable()`, running tethered

**Comment by edjubuh** [comment]
@baylessj, can you test no standalone mode with a joystick (one of USB A-A or VEXNet - don't care which)

**Comment by HotelCalifornia** [comment]
Bonus points if you test them both

**Comment by edjubuh** [comment]
[kernel-2.11.2.zip](https://github.com/purduesigbots/pros/files/1093394/kernel-2.11.2.zip)

To test, extract the files to `%AppData%\PROS\pros-mainline\kernel-2.11.2` such that the file `%AppData%\PROS\pros-mainline\kernel-2.11.2\template.pros` exists. For Linux and MacOS, the same applies but the folder is `~/.pros` instead of `%AppData%\pros`

File: src/supervisor.c [hunk 3]
```diff
@@ -133,6 +133,12 @@ void _svNextByte() {
 	SPI1->DR = spiBufferTX[idx];
 }
 
+// standaloneModeEnable - enable standalone operation
+void standaloneModeEnable() {
+	// set flag in outgoing SPI buffer to enable standlone mode
+	SV_OUT->flags |= 1;
```
Comment (jbuschjr) [inline src/supervisor.c]: Are we aware currently of what the remaining bits of this `flags` byte (I assume it to be `uint8_t`) do?

Comment (edjubuh) [inline src/supervisor.c]: No, we'll have to take a look at the other bit values during the fall semester.

**Pull request merged**